### ISSUE TITLE
Base href is ignored in the built-in implicit callback component

### ISSUE THREAD

**Issue by albertogarza** [okta/okta-oidc-js#745]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [x] okta-angular
- [ ] oidc-middleware
- [ ] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [x] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

When the Angular app is built with the flag `--base-href` set to a value other than `/`, the implicit callback component (`callback.component.ts`) is completely missing this base href value when redirecting the user after a successful log in attempt.

## Expected behavior

This callback component should take the base href value into consideration.

## Minimal reproduction of the problem with instructions

Run the ng build command with `--base-href=/test/` and run a quick Okta log in test.

## Extra information about the use case/user story you are trying to implement

Upon a close look at the source code, I discovered that the `callback.component.ts` file is using the following JavaScript line of code to accomplish the redirect:

```javascript
        window.location.replace(fromUri);
```

The value for `fromUri` is actually stored previously before the user is even redirected to log in while executing the route guard `okta.guard.ts`:

```typescript
    /**
     * Store the current path
     */
    this.oktaAuth.setFromUri(state.url);
```

As a workaround, I implemented my own version of the callback component with the following slight change:

```typescript
import { Component, OnInit } from '@angular/core';
import { OktaAuthService } from '@okta/okta-angular';
import { Router } from '@angular/router';

@Component({
  template: `
    <div>{{ error }}</div>
  `
})
export class CallbackComponent implements OnInit {
  error: string;

  constructor(private okta: OktaAuthService, private router: Router) {}

  async ngOnInit() {
    /**
     * Handles the response from Okta and parses tokens.
     */
    return this.okta
      .handleAuthentication()
      .then(() => {
        /**
         * Navigate back to the saved uri, or root of application.
         */
        const fromUri = this.okta.getFromUri();
        // window.location.replace(fromUri);    <-- OLD CODE
        this.router.navigate([fromUri]);     // <-- NEW CODE 
      })
      .catch(e => {
        this.error = e.toString();
      });
  }
}
```

I believe this should be fixed for this project as well.

## Environment

- Package Version: 
```
    "@okta/okta-angular": "^2.0.0",
    "@okta/okta-signin-widget": "^3.8.1",
```
- Browser: Chrome
- OS: Mac OS
- Node version (`node -v`): 10.17
- Other:

**Comment by swiftone** [comment]
Internal ref: OKTA-295172

**Comment by aarongranick-okta** [comment]
@albertogarza We've investigated this issue and are determining the best way to provide support for `--base-href` option without breaking existing implementations. Most likely, we will add full support in the next major version release. 

In the meantime, your workaround of providing a custom callback component seems to work well. There are a few other details to be aware of:

- `setFromUri` is called on protected routes by `okta.guard.ts`, in this case it is called with a relative route path without the base href. Your component handles converting this route path to the correct URL, including base.
- `setFromUri` will not be called if you  call `loginRedirect` directly with no parameters. If `setFromUri` has not been called, `getFromUri` will return `window.location.origin` which will not include the base href. If you call `loginRedirect` directly (from a "signin button" for example), you should pass the relative URL path as the first parameter.

You probably already have these set already, but the values for `redirectUri` and `postLogoutRedirectUri` should both contain the base href path. This is not done automatically by the `--base-href` option, it will have to be configured. These URLs also need to be added to the application settings in the Okta admin UI.

**Comment by swiftone** [comment]
Support for base href was added/fixed in okta-angular 3.0.1 - https://github.com/okta/okta-angular/releases/tag/okta-angular-3.0.1

Please let us know if this does not address your issue.
